Main/ripple.sh

```diff
@@ -0,0 +1,358 @@
+			./configure --enable-optimizations ; make ; make install
+		)
+	}
+
+	python3_5
+	python3_6
+}
+
+# FIXME: check all dependencies, add/remove if needed for all supported systems.
+# Install necessary dependencies required for lets, pep.py, hanayo, go, old-frontend, mysql. (Used APT)
+dependencies() {
+	printf '%s\n' "Starting To Install Required/Necessary Dependencies..." ; sleep 2
+
+	if command -v apt >/dev/null; then
+		package_manager="apt"
+		printf '%s\n' "Using Package Manager: $package_manager"
+		"$package_manager" update ; "$package_manager" upgrade -y
+		"$package_manager" install build-essential gcc g++ \
+		git tmux nginx wget gnupg mysql-server redis-server nginx checkinstall cython \
+		php-fpm composer php7.0-mbstring php7.0-curl php-mysql -y
+
+		# Golang1.13+ for Hanayo & rippleapi
+		(
+			mkdir builds ; cd builds || exit
+			wget https://dl.google.com/go/go1.13.3.linux-amd64.tar.gz
+			tar -xvf go1.13.3.linux-amd64.tar.gz ; mv go /usr/local
+			export GOROOT=/root/go
+			export PATH=$GOPATH/bin:$GOROOT/bin:$PATH
+		)
+		# FIXME: it's ugly way to do. Provide me better way of this implementation
+		python3 ; "$package_manager" install python3-dev
+
+		printf '%s\n' "Done Installing all the necessary Dependencies with [$package_manager]" ; sleep 1
+
+	elif ! command -v apt >/dev/null; then
+    		printf '%s\n' "apt is not executable on this system!"
+
+	elif command -v pacman >/dev/null; then
+		package_manager="pacman"
+		printf '%s\n' "Using Package Manager: $package_manager"
+		"$package_manager" -Syyu --noconfirm
+		"$package_manager" -S --noconfirm tmux git wget gnupg mysql redis nginx cython php composer go
+		python3
+
+	elif ! command -v pacman >/dev/null; then
+		printf '%s\n' "pacman is not executable on this system!"
+
+	else
+    	printf '%s\n' "Unexpected Error!"
+	fi
+
+}
+# FIXME: Add proper cloning with checks & sanitization
+main_dir() {
+	# Creating Master Directory (where all The Repositories will be cloned)
+	while [ -z "$targetDir" ]; do
+		read -r "Enter Master Directory: " targetDir
+	done
+
+	if [ -n "$targetDir" ]; then
+		master_dir="$(pwd)/$targetDir"
+		read -r "Create Master Directory: $master_dir ? y/n " confirmation
+		if [ "$confirmation" = "y" ]; then
+			mkdir "$master_dir"
+			if [ -d "$master_dir" ]; then
+				chmod -R 777 "$master_dir"
+				export directory="$master_dir"
+			    	printf "%s has been created!\n" "$master_dir"
+			else
+				printf '%s\n' "Failed to create Directory: $master_dir"
+			fi
+		else
+			printf '%s\n' "Directory: $master_dir wasn't created! Exiting...\n"
+			exit 1
+		fi
+	fi
+
+}
+
+# peppy is the backend of osu/bancho, starting from client login, it handles most of the stuff.
+peppy () {
+
+	task="pep.py"
+	# FIXME: Add proper cloning with checks & sanitization including automated configuration
+	printf '%s\n' "Cloning and Setting up pep.py $task" ; sleep 2
+	(
+		if [ -n "$directory" ]; then
+			cd "$directory" || exit
```

Comment:
avoid using cd where possible, git does have a `git clone ... path/to/targetdir` method